include/mgp_mock.py

```diff
@@ -0,0 +1,1011 @@
+
+    params = tuple(sig.parameters.values())
+    if not params or not params[0].annotation is Messages:
+        if not len(params) == 2 or not params[1].annotation is Messages:
+            raise NotImplementedError("Valid signatures for transformations are (TransCtx, Messages) or (Messages)")
+
+    if params[0].annotation is TransCtx:
+
+        @wraps(func)
+        def wrapper(ctx, messages):
+            return func(ctx, messages)
+
+    else:
+
+        @wraps(func)
+        def wrapper(_, messages):
+            return func(messages)
+
+    return wrapper
+
+
+class FuncCtx:
+    __slots__ = "_graph"
+
+    def __init__(self, graph):
+        if not isinstance(graph, nx.MultiDiGraph):
+            raise TypeError(f"Expected 'networkx.classes.multidigraph.MultiDiGraph', got '{type(graph)}'")
+
+        self._graph = Graph(graph)
+
+    def is_valid(self) -> bool:
+        return True
+
+
+def function(func: typing.Callable):
+    raise_if_does_not_meet_requirements(func)
+
+    sig = inspect.signature(func)
+
+    params = tuple(sig.parameters.values())
+    if params and params[0].annotation is FuncCtx:
+
+        @wraps(func)
+        def wrapper(ctx, *args):
+            graph = nx.freeze(ctx.graph._graph.nx)
+            return func(FuncCtx(graph), *args)
+
+    else:
+
+        @wraps(func)
+        def wrapper(*args):
+            return func(*args)
+
+    return wrapper
+
+
+def _wrap_exceptions():
+    def wrap_function(func):
+        @wraps(func)
+        def wrapped_func(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except _mgp_mock.LogicErrorError as e:
+                raise LogicErrorError(e)
+            except _mgp_mock.ImmutableObjectError as e:
+                raise ImmutableObjectError(e)
+
+        return wrapped_func
+
+    def wrap_prop_func(func):
+        return None if func is None else wrap_function(func)
+
+    def wrap_member_functions(cls: type):
+        for name, obj in inspect.getmembers(cls):
+            if inspect.isfunction(obj):
+                setattr(cls, name, wrap_function(obj))
+            elif isinstance(obj, property):
+                setattr(
+                    cls,
+                    name,
+                    property(
+                        wrap_prop_func(obj.fget),
+                        wrap_prop_func(obj.fset),
+                        wrap_prop_func(obj.fdel),
+                        obj.__doc__,
+                    ),
+                )
+
+    def defined_in_this_module(obj: object):
+        return getattr(obj, "__module__", "") == __name__
+
+    module = sys.modules[__name__]
+    for name, obj in inspect.getmembers(module):
+        if not defined_in_this_module(obj):
+            continue
+        if inspect.isclass(obj):
+            wrap_member_functions(obj)
+        elif inspect.isfunction(obj) and not name.startswith("_"):
+            setattr(module, name, wrap_function(obj))
+
+
```

Comment:
I think Logger class is missing, we can also add that and provide api to logging

Reply:
Assuming that the Logger is a Memgraph-specific logger, why should it feature in the mock API? The mock API is supposed to run independently of Memgraph.

Reply:
I was thinking to have idea how it would work with logging, but we don't have to add that.